src/opnsense/www/js/opnsense_theme.js

```diff
@@ -65,35 +91,64 @@ $(document).ready(function () {
         /* navigation toggle */
         $(".toggle-sidebar").click(function () {
             opnsense_sidebar_toggle(true);
-            $(this).blur();
+            $(this).blur();			
         });
-
-        /* sidebar mouseevents */
-        $("#navigation").hover(function () {
+		/* sidebar mouseenter */
+        $("#navigation").mouseenter(function () {
             if ($("#navigation").hasClass("col-sidebar-left")) {
-                $("#navigation > div > nav > #mainmenu > div > div").on({
-                    mouseout: function() {$("#navigation.col-sidebar-left").css("width", "70px"); },
-                    mouseover: function() {$("#navigation.col-sidebar-left").css("width", "380px"); }
-                });
-                $("#navigation > div > nav > #mainmenu > div > a").on({
-                    mouseout: function() {$("#navigation.col-sidebar-left").css("width", "70px"); },
-                    mouseover: function() {$("#navigation.col-sidebar-left").css("width", "380px"); }
-                });
-            }
-        });
-
-        /* on resize - toggle sidebar / main navigation */
+                transition_duration(0);
+				layer1_a.on({
+					mouseenter: function() {	$("#navigation.col-sidebar-left").css("width", "380px");
+												$(this).trigger("click");},
+					mouseleave: function() { 	$("#navigation.col-sidebar-left").css("width", "70px");},
+					mousedown: function() { 	$(this).trigger("click");}
+							});
+				
+				layer2_a.on({
+					mouseenter: function() {	$("#navigation.col-sidebar-left").css("width", "380px");
+												$(this).trigger("click");},
+					mouseleave: function() { 	$("#navigation.col-sidebar-left").css("width", "70px");},
+					mousedown: function() { 	$(this).trigger("click");}	
+							});
+				
+				layer1_div.on({
+					mouseenter: function() { 	$("#navigation.col-sidebar-left").css("width", "380px");
+												$(this).trigger("click");},
+					mouseleave: function() { 	$("#navigation.col-sidebar-left").css("width", "70px");},
+							});
+				
+				layer2_div.on({
+					mouseenter: function() { 	$("#navigation.col-sidebar-left").css("width", "380px");
+												$(this).trigger("click");},
+					mouseleave: function() { 	$("#navigation.col-sidebar-left").css("width", "70px");},
+							});														
```

Comment:
Two issues:
* https://en.wikipedia.org/wiki/Don%27t_repeat_yourself
* this code layout is awful